Sources/MessageInputBar.swift

```diff
@@ -24,133 +24,434 @@
+            textViewLayoutSet?.right?.constant = -textViewPadding.right
+            bottomStackViewLayoutSet?.top?.constant = textViewPadding.bottom
+        }
     }
-
-    convenience public init() {
+    
+    open lazy var sendButton: InputBarButtonItem = { [unowned self] in
+        return InputBarButtonItem()
+            .configure {
+                $0.size = CGSize(width: 52, height: 36)
+                $0.spacing = .fixed(4)
+                $0.isEnabled = false
+                $0.title = "Send"
+                $0.titleLabel?.font = UIFont.preferredFont(forTextStyle: .headline)
+            }.onTextViewDidChange({ (item, textView) in
+                item.isEnabled = !textView.text.isEmpty
+            }).onTouchUpInside {
+                $0.inputBarAccessoryView?.didSelectSendButton()
+        }
+        }()
+    
+    /// The anchor contants used by the UIStackViews and InputTextView to create padding within the InputBarAccessoryView
+    open var padding: UIEdgeInsets = UIEdgeInsets(top: 6, left: 12, bottom: 6, right: 12) {
+        didSet {
+            updateViewContraints()
+        }
+    }
+    
+    override open var intrinsicContentSize: CGSize {
+        let sizeToFit = textView.sizeThatFits(CGSize(width: textView.bounds.width, height: .greatestFiniteMagnitude))
+        var heightToFit = sizeToFit.height.rounded() + padding.top + padding.bottom
+        if heightToFit >= maxHeight {
+            textView.isScrollEnabled = true
+            heightToFit = maxHeight
+        } else {
+            textView.isScrollEnabled = false
+        }
+        let size = CGSize(width: bounds.width, height: heightToFit)
+        if previousIntrinsicContentSize != size {
+            delegate?.inputBar?(self, didChangeIntrinsicContentTo: size)
+        }
+        previousIntrinsicContentSize = size
+        return size
+    }
+    
+    /// The maximum intrinsicContentSize height. When reached the delegate 'didChangeIntrinsicContentTo' will be called.
+    open var maxHeight: CGFloat = 288 {
+        didSet {
+            invalidateIntrinsicContentSize()
+        }
+    }
+    
+    /// The fixed widthAnchor constant of the leftStackView
+    open var leftStackViewWidthContant: CGFloat = 0 {
+        didSet {
+            leftStackViewLayoutSet?.width?.constant = leftStackViewWidthContant
+        }
+    }
+    
+    /// The fixed widthAnchor constant of the rightStackView
+    open var rightStackViewWidthContant: CGFloat = 52 {
+        didSet {
+            rightStackViewLayoutSet?.width?.constant = rightStackViewWidthContant
+        }
+    }
+    
+    /// The InputBarItems held in the leftStackView
+    open var leftStackViewItems: [InputBarButtonItem] = []
+    
+    /// The InputBarItems held in the rightStackView
+    open var rightStackViewItems: [InputBarButtonItem] = []
+    
+    /// The InputBarItems held in the bottomStackView
+    open var bottomStackViewItems: [InputBarButtonItem] = []
```

Comment:
Maybe we want the storage for these buttons to be `private(set)` so they're not able to be modified directly